### PR TITLE
http2: fix test-http2-binding after round 11

### PR DESCRIPTION
**Pull request by mcollina**
As titled, there was a broken test.

### CODE REVIEW

**Review by danbev — APPROVED** [review]
All tests pass for me now.

File: test/parallel/test-http2-binding.js [hunk 1]
```diff
@@ -25,10 +25,9 @@ assert.strictEqual(http2.nghttp2ErrorString(-517),
 const check = Buffer.from([0x00, 0x01, 0x00, 0x00, 0x10, 0x00, 0x00, 0x04,
                            0x00, 0x00, 0xff, 0xff, 0x00, 0x05, 0x00, 0x00,
                            0x40, 0x00, 0x00, 0x02, 0x00, 0x00, 0x00, 0x01]);
-const val = http2.packSettings(http2.getDefaultSettings({}));
+const val = http2.packSettings(require('http2').getDefaultSettings());
 assert.deepStrictEqual(val, check);
 
-assert.doesNotThrow(() => assert(require('http2').getDefaultSettings()));
```
Comment (robertkowalski) [inline test/parallel/test-http2-binding.js:31]: just curious: is it that after a `packSettings` call `getDefaultSettings()` should throw?

File: test/parallel/test-http2-binding.js [hunk 2]
```diff
@@ -13,7 +13,7 @@ assert(http2.SessionShutdownWrap);
 assert.strictEqual(typeof http2.Http2Session, 'function');
 assert.strictEqual(typeof http2.SessionShutdownWrap, 'function');
 
-const settings = http2.getDefaultSettings({});
+const settings = require('http2').getDefaultSettings();
```
Comment (robertkowalski) [inline test/parallel/test-http2-binding.js:16]: why is require used here, I believe the module cache delivers ithe module anyway from the cache?

**Review by jasnell — APPROVED** [review]

**Comment by robertkowalski** [comment]
makes all sense now, thanks for the explanations!

File: test/parallel/test-http2-binding.js (hunk 1 above)
Comment (jasnell) [reply test/parallel/test-http2-binding.js:31]: no, those are separate operations. `packSettings()` returns a base64 encoded string with the settings, `getDefaultSettings()` returns the default settings object.

**Comment by jasnell** [comment]
Note that this test could likely use some general refactoring. When I wrote it initially the goal was just to do a quick rudimentary test that things we generally ok. I was not writing it for quality.

**Comment by jasnell** [comment]
Landed!

**Pull request closed without merging**